File: Domain/Classes/Account.cs
namespace Domain.Classes;

public sealed class Account
{
    public User User { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string Password { get; set; } = null!;
}
public sealed class CreateAccount
{
    public CreateUser User { get; set; } = null!;
    public string Email { get; set; } = null!;
    public string Password { get; set; } = null!;
}

File: domain/entities/account.rs
pub mod create_account;
pub mod create_partial_account;

use crate::entities::user::User;

#[derive(Debug, Clone)]
pub struct Account {
  pub user: User,
  pub password: Password,
}

#[derive(Debug, Clone)]
#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
pub struct Password(pub String);
